File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/csharp/PrimitiveTypesExamples.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace ClipboardExamples
{
    public class PrimitiveTypesExamples
    {
        // <PrimitiveTypesExample>
        public static void PrimitiveTypesExample()
        {
            // Numeric types
            Clipboard.SetData("MyInt", 42);
            Clipboard.SetData("MyDouble", 3.14159);
            Clipboard.SetData("MyDecimal", 123.45m);

            // Text and character types
            Clipboard.SetData("MyString", "Hello World");
            Clipboard.SetData("MyChar", 'A');

            // Boolean and date/time types
            Clipboard.SetData("MyBool", true);
            Clipboard.SetData("MyDateTime", DateTime.Now);
            Clipboard.SetData("MyTimeSpan", TimeSpan.FromMinutes(30));

            // Later retrieval with type safety
            if (Clipboard.TryGetData("MyTimeSpan", out TimeSpan value))
            {
                Console.WriteLine($"Clipboard value is: {value}");
            }
        }
        // </PrimitiveTypesExample>

        // <CollectionsExample>
        public static void CollectionsExample()
        {
            // Arrays of primitive types
            int[] numbers = { 1, 2, 3, 4, 5 };
            Clipboard.SetData("NumberArray", numbers);

            double[] coordinates = { 1.0, 2.5, 3.7 };
            Clipboard.SetData("Coordinates", coordinates);

            // Generic lists
            List<int> intList = new List<int> { 10, 20, 30 };
            Clipboard.SetData("IntList", intList);

            // Retrieval maintains type safety
            if (Clipboard.TryGetData("NumberArray", out int[] retrievedNumbers))
            {
                Console.WriteLine($"Numbers: {string.Join(", ", retrievedNumbers)}");
            }
        }
        // </CollectionsExample>
    }
}

File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/vb/PrimitiveTypesExamples.vb
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms

Namespace ClipboardExamples
    Public Class PrimitiveTypesExamples
        ' <PrimitiveTypesExample>
        Public Shared Sub PrimitiveTypesExample()
            ' Numeric types
            Clipboard.SetData("MyInt", 42)
            Clipboard.SetData("MyDouble", 3.14159)
            Clipboard.SetData("MyDecimal", 123.45D)

            ' Text and character types
            Clipboard.SetData("MyString", "Hello World")
            Clipboard.SetData("MyChar", "A"c)

            ' Boolean and date/time types
            Clipboard.SetData("MyBool", True)
            Clipboard.SetData("MyDateTime", DateTime.Now)
            Clipboard.SetData("MyTimeSpan", TimeSpan.FromMinutes(30))

            ' Later retrieval with type safety
            Dim value As TimeSpan

            If Clipboard.TryGetData("MyTimeSpan", value) Then
                Console.WriteLine($"Clipboard value is: {value}")
            End If
        End Sub
        ' </PrimitiveTypesExample>

        ' <CollectionsExample>
        Public Shared Sub CollectionsExample()
            ' Arrays of primitive types
            Dim numbers As Integer() = {1, 2, 3, 4, 5}
            Clipboard.SetData("NumberArray", numbers)

            Dim coordinates As Double() = {1.0, 2.5, 3.7}
            Clipboard.SetData("Coordinates", coordinates)

            ' Generic lists
            Dim intList As New List(Of Integer) From {10, 20, 30}
            Clipboard.SetData("IntList", intList)

            ' Retrieval maintains type safety
            Dim retrievedNumbers As Integer() = Nothing

            If Clipboard.TryGetData("NumberArray", retrievedNumbers) Then
                Console.WriteLine($"Numbers: {String.Join(", ", retrievedNumbers)}")
            End If
        End Sub
        ' </CollectionsExample>
    End Class
End Namespace
